contracts/ExponentialReverseDutchAuction.sol

```diff
@@ -0,0 +1,199 @@
+        }else{
+            require(IERC20(auctionedToken).balanceOf(msg.sender)>=auctionedTokenIdOrAmount,"Insufficient balance");
+            SafeERC20.safeTransferFrom(IERC20(auctionedToken),msg.sender,address(this),auctionedTokenIdOrAmount);
+        }
+
+        uint256 deadline=block.timestamp+duration;
+
+        auctions[auctionCounter] = AuctionData({
+            id: auctionCounter,
+            name:name,
+            description:description,
+            imgUrl:imgUrl,
+            auctioneer: msg.sender,
+            auctionType: auctionType,
+            auctionedToken: auctionedToken,
+            auctionedTokenIdOrAmount: auctionedTokenIdOrAmount,
+            biddingToken: biddingToken,
+            startingPrice: startingPrice,
+            availableFunds: 0,
+            reservedPrice: reservedPrice,
+            decayFactor: decayFactor,
+            winner: msg.sender,
+            deadline: deadline,
+            duration: duration,
+            isClaimed: false
+        });
+
+        emit AuctionCreated(
+            auctionCounter++,
+            name,
+            description,
+            imgUrl,
+            msg.sender,
+            auctionType,
+            auctionedToken,
+            auctionedTokenIdOrAmount,
+            biddingToken,
+            startingPrice,
+            reservedPrice,
+            decayFactor,
+            deadline
+        );
+    }
+
+    function getDecayValue(uint256 x) internal view returns (uint256){
+        if(x >= 61 * 1e3) {
+            return 0;
+        }else{
+            uint256 scaledPower=x/1e3;
+            uint256 remainder=x%1e3;
+            
+            if(remainder==0) return decayLookup[scaledPower];
+
+            uint256 higherValue=decayLookup[scaledPower];
+            uint256 lowerValue=scaledPower<61? decayLookup[scaledPower+1] : 0;
+            return higherValue - ((higherValue-lowerValue)*remainder)/1e3;  //Linear interpolation
+        }
+    }
+
+    function getCurrentPrice(uint256 auctionId) public view validAuctionId(auctionId) returns (uint256) {
+        AuctionData storage auction=auctions[auctionId];
+        require(block.timestamp<auction.deadline,"Auction has ended");
+        require(!auction.isClaimed,"Auction has ended");
+
+        uint256 timeElapsed=block.timestamp-(auction.deadline-auction.duration);
+        uint256 x=timeElapsed * auction.decayFactor;
+        uint256 decayValue=getDecayValue(x);
+
+        // price(t) = reservedPrice + (startingPrice - reservedPrice) * 2^(-timeElapsed*decayFactor)
+        return auction.reservedPrice + ((auction.startingPrice-auction.reservedPrice)*decayValue)/1e18; 
+    }
+
+    //placeBid function is not required in this auction type as the price is determined by the auctioneer and the auctioneer can withdraw the item on placing the bid only
+    function withdrawItem(uint256 auctionId,uint256 bidAmount) external validAuctionId(auctionId) {
```

Comment:
We don't need `bidAmount` here, since the bidder will pay `currentPrice` anyway.